src/devices/bus/coco/meb_rtime.cpp

```diff
@@ -0,0 +1,169 @@
+// license:BSD-3-Clause
+// copyright-holders:tim lindner
+/***************************************************************************
+
+    meb_rtime.cpp
+
+    Real time clock for Disto mini expansion bus
+
+    Includes a Centronics parallel port interface
+
+***************************************************************************/
+
+#include "emu.h"
+#include "meb_rtime.h"
+#include "machine/msm6242.h"
+#include "bus/centronics/ctronics.h"
+
+#include "meb_intrf.h"
+
+
+// ======================> disto_rtime_device
+
+namespace
+{
+	class disto_rtime_device
+		: public device_t
+		, public device_distomeb_interface
+	{
+		public:
+			// construction/destruction
+			disto_rtime_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+		protected:
+			// device-level overrides
+			virtual void device_start() override;
+			virtual void device_add_mconfig(machine_config &config) override;
+			virtual u8 meb_read(offs_t offset) override;
+			virtual void meb_write(offs_t offset, u8 data) override;
+
+		private:
+			DECLARE_WRITE_LINE_MEMBER(busy_w);
+
+			required_device<msm6242_device> m_rtc;
+			u8 m_rtc_address;
+			u8 m_double_write;
+			required_device<centronics_device> m_centronics;
+			required_device<output_latch_device> m_latch;
+			int m_centronics_busy;
```

Comment:
You should use an explicitly sized integer for this as it should be registered for save states.